src/components/comment.jsx

```diff
@@ -0,0 +1,144 @@
+import React, { useState } from 'react';
+import Edit from './edit';
+import Delete from './delete'
+
+const Comment = (props) => {
+  const [replyClicked, setReplyClicked] = useState(false);
+  const [reply, setReply] = useState('');
+  const [comments, setComments] = useState(props.comments);
+
+  const handleReplySubmit = (e) => {
+    e.preventDefault();
+    if (reply === '') {
+      return;
+    }
+
+    const replyData = {
+      id: 0,
+      content: reply,
+      createdAt: 'now',
+      score: 0,
+      replyingTo: '',
+      user: {
+        image: {
+          png: '',
+          webp: '',
+        },
+        username: '',
+      },
+    };
+    let newComments = props.comments.map((comment, index) => {
+      if (comment.id === props.comment.id) {
+        comment.replies.push(replyData);
+      }
+      return comment;
+    });
+
+    props.setComments(newComments);
+    setReplyClicked(false);
+    setReply('');
+  };
+
+  const handlePlus = () => {
+    const newComments = props.comments.map((comment, index) => {
+      if (comment.id === props.comment.id) {
+        comment.score += 1;
+      }
+      return comment;
+    });
+    props.setComments(newComments);
+  };
+
+  const handleMinus = () => {
+    const newComments = props.comments.map((comment, index) => {
+      if (comment.score === 0) {
+        return comment;
+      }
+      if (comment.id === props.comment.id) {
+        comment.score -= 1;
+      }
+      return comment;
+    });
+    props.setComments(newComments);
+  };
+
+  return (
+    <div className="bg-white m-2 flex flex-row p-2" id="comment-container">
+      <div id="score-container" className="bg-gray-100 text-purple-900 m-2 rounded-lg h-[130px] w-[30px] flex flex-col justify-center">
+      
+        <button
+        id={`plus-button-${props.comment.id}`}
+        onClick={() => handlePlus()}
```

Comment:
if the function does not pass in an event or an argument you can just do 
```onClick={handlePlus}```

likewise for minus

Reply:
done :)